### PR TITLE
Serialized Feedback Editor

### PR DESCRIPTION
**Pull request by mtschoen-unity**
**Purpose of this PR**
There was a need to clear out the serialized data that stores the number of visible feedback presentations while testing. For example, if I want to show the system to a new user, I might want them to have a "fresh start," or I might want to just enable feedback for one or two features.

This gives us a way of viewing and editing the serialized data for feedback.

**Testing status**

Tested on Vive and Rift in various scenes.  Tested starting EXR without activating the controllers, and with activating controllers but not engaging feedback. Found and fixed a bug where feedback data could be cleared, because we were always returning a SerializedFeedback, even if the nodes themselves were null.

**Technical risk**

Low: Only changes to actual EXR stuff were fixes/refactors. Adds an editor window.

**Comments to reviewers**

I had to expose a bunch of stuff as internal to get static access or access without EXR running. There is a little bit of code duplication as a result, and because Feedback Module is designed around being instantiated and working with EXR systems.

### CODE REVIEW

File: Scripts/Proxies/Data/SerializedProxyFeedback.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Contains per-request persistent data
+    /// </summary>
+    [Serializable]
+    class RequestData
+    {
+        readonly Action m_OnBecameVisible;
```
Comment (dunity) [inline Scripts/Proxies/Data/SerializedProxyFeedback.cs]: Move below properties

File: Editor/ProxyFeedbackEditor.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+using System.Collections.Generic;
+using UnityEditor.Experimental.EditorVR.Modules;
+using UnityEditor.Experimental.EditorVR.Proxies;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.UI
+{
+    sealed class ProxyFeedbackEditor : EditorWindow
+    {
+        Dictionary<Type, SerializedProxyFeedback> m_SerializedFeedback = new Dictionary<Type, SerializedProxyFeedback>();
```
Comment (dunity) [inline Editor/ProxyFeedbackEditor.cs]: make readonly

File: Editor/ProxyFeedbackEditor.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            }
+        }
+
+        static void ClearData()
+        {
+            var preferences = SerializedPreferencesModule.DeserializePreferences(Core.EditorVR.serializedPreferences);
+            if (preferences == null)
+                return;
+
+            foreach (var kvp in new Dictionary<Type, SerializedPreferencesModule.SerializedPreferenceItem>(preferences.items))
```
Comment (stella3d) [inline Editor/ProxyFeedbackEditor.cs]: is it necessary to create a new dict ?  `preferences.items` is already a `Dictionary<Type, SerializedPreferenceItem>`

Comment (mtschoen-unity) [reply Editor/ProxyFeedbackEditor.cs]: We are removing elements

**Comment by mtschoen-unity** [comment]
Oh yeah I forgot to ask/mention: is anyone else seeing errors all over ProxyFeedbackEditor within Visual Studio? It seems like it's ignoring the "InternalsVisibleTo" that we set up to access Engine stuff from an Editor class. If everyone else is getting it, I'll report a bug

File: Scripts/Proxies/Data/SerializedProxyFeedback.cs (hunk 1 above)
Comment (mtschoen-unity) [reply Scripts/Proxies/Data/SerializedProxyFeedback.cs]: This is addressed. Not sure why it's not hidden.

**Comment by mtschoen-unity** [comment]
Manual merge successful. Performing Github merge

**Pull request merged**